docs/developer/dpf_selector.md

```diff
@@ -0,0 +1,147 @@
+# Technical concept DPF Selector
+
+This document presents the technical concept for the "DataPlaneFramework Selector", henceforth "DPF Selector" or "DPFS" for short. 
+In typical clustered environments (Kubernetes) we anticipate a multitude of connector runtimes and DPF runtimes, scaling independently from each other, depending
+on a particular use case. Also, the DPFs may have different configurations, so they may not even be in the same replica set (in Azure this would be Availability Sets vs. Scale Sets).
+
+Further, we assume that every connector has access to multiple DPF runtimes, even if they are not homogenous replicas. For example, some deployments may choose to have one dedicated DPF per geographical region, or even per `DataSink`/`DataSource` pair.
+
+Whenever a data transfer request hits the connector, it eventually will delegate the transfer to the DFP. For that, it needs to know _which DPF to delegate to_. In simple deployments, there may just be one single DPF, so the DPF selector is essentially a preconfigured value, but in more complex scenarios sophisticated algorithms may be employed to select the best-fitting DPF. 
+
+This is not dissimilar to how load-balancers work, one key difference being that the DPF selector will distribute the request based on not just the load situation, but also based on request metadata or even functional requirements (e.g. "must be able to store in Azure BlobStore").
+
+Another key difference to (most) load balancers is the modularity and flexibility of the DPF selector: it can be embedded into another process (e.g. a connector), standalone, it can handle different storage and API technologies, it itself could theoretically run clustered, etc.
+
+## Terminology
+- DPF: one DataPlaneFramework runtime instance
+- CP: one ControlPlane instance, often used synonymous to "connector"
+- connector: one connector runtime
+- selector: one DPF selector runtime, either embedded or standalone
+- DataSource: one particular implementation that the DPF uses to fetch data from (e.g. Amazon S3)
+- DataSink: one particular implementation of a data destination (e.g. AzureBlob)
+- Seeding: the process of pre-populating a database. Often used synonymous to "loading".
+- Feature pack: a collection of Java modules that fulfill one particular purpose, typically accompanied by a BOM (bill-of-materials)
+
+## Proposed solution
+
+We propose to create another "feature pack" in the EDC repository, essentially a meta module named `:extensions:data-plane:data-plane-selector`, which will contain (at least) the following submodules:
```

Comment:
I am a bit perplex here. 
The DPF selector will be called by the **Control Plane** (either remotely of directly with an embedded runtime). So for me it should not belong to the `data-plane` module as it contains only stuff to be embedded within the DPF runtime. 

In current implementation, all the stuff related to the DPF but that belong to the Control Plane runtime is in the `data-plane-transfer` module. Worth noting that this later module has a dependency on `core-transfer` while the `data-plane` module doesn't (shouldn't). 

So my recommendation would be to put this `data-plane-selector` submodule within `data-plane-transfer` instead of `data-plane` module. 

WDYT?

Reply:
Sounds like having the `data-plane-selector` at the same directory level as `data-plane` and `data-plane-transfer` is the appropriate solution?

Reply:
Yup.